### DISCUSSION TITLE
Allow `.` in `slug` field

### DISCUSSION THREAD

**Discussion by notramo** [martenframework/marten#166] [Ideas]
Usernames often contain `.` characters (see email addresses, or Instagram, TikTok usernames).
If the `slug` field is kept as-is, it would be useful to add a `username` field type.

**Comment by ellmetha** [comment]
Hey! 👋 

Slugs can't contain `.` characters technically! I would suggest using a simple `string` field with an associated validation rule where you validate that the field value matches a regex corresponding to your username format requirements.

I am not sure a `username` field would fit within the framework because the requirements around a "username" value may vary easily from one project to another. Moreover, I don't think this is a common enough use case to make this part of the framework (nowadays most user models in web apps don't need a public username, and authentication is usually done using an email address for example).
